### PR TITLE
SONAR-7692 replace columns with IDs by UUIDs in table RESOURCE_INDEX

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/version/v60/PopulateUuidColumnsOfResourceIndex.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  public void execute(Context context) throws SQLException {
+    Map<Long, String> componentUuidById = buildComponentUuidMap(context);
+    if (componentUuidById.isEmpty()) {
+      return;
+    }
+
+    populateUuidColumns(context, componentUuidById);
+  }
+
+  private Map<Long, String> buildComponentUuidMap(Context context) throws SQLException {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/version/v60/PopulateUuidColumnsOfResourceIndex.java]: MAJOR Make "buildComponentUuidMap" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1203_clean_orphelin_rows_in_resource_index.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+#
+# SonarQube 6.0
+#
+class CleanOrphelinRowsInResourceIndex < ActiveRecord::Migration
```
Comment (simonbrandhof) [inline server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1203_clean_orphelin_rows_in_resource_index.rb]: "orphan" :-)

Comment (simonbrandhof) [inline server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1203_clean_orphelin_rows_in_resource_index.rb]: what about executing this step before updates of UUIDs ?

File: sonar-db/src/main/java/org/sonar/db/version/v60/DropIdColumnsFromResourceIndex.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+public class DropIdColumnsFromResourceIndex extends DdlChange {
+
+  private static final String TABLE_RESOURCE_INDEX = "resource_index";
+
+  private final Database db;
+
+  public DropIdColumnsFromResourceIndex(Database db) {
+    super(db);
+    this.db = db;
```
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/version/v60/DropIdColumnsFromResourceIndex.java]: no need to store `db` twice. The field can be removed and calls replaced by the protected method `getDatabase()`

File: sonar-db/src/main/java/org/sonar/db/version/v60/MakeUuidColumnsNotNullOnResourceIndex.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+public class MakeUuidColumnsNotNullOnResourceIndex extends DdlChange {
+
+  private static final String TABLE_RESOURCE_INDEX = "resource_index";
+
+  private final Database db;
+
+  public MakeUuidColumnsNotNullOnResourceIndex(Database db) {
+    super(db);
+    this.db = db;
```
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/version/v60/MakeUuidColumnsNotNullOnResourceIndex.java]: no need to store db twice. The field can be removed and calls replaced by the protected method getDatabase()

File: sonar-db/src/main/java/org/sonar/db/version/v60/AddUuidColumnsToResourceIndex.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+public class AddUuidColumnsToResourceIndex extends DdlChange {
+
+  private static final String TABLE_RESOURCE_INDEX = "resource_index";
+
+  private final Database db;
+
+  public AddUuidColumnsToResourceIndex(Database db) {
+    super(db);
+    this.db = db;
```
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/version/v60/AddUuidColumnsToResourceIndex.java]: no need to store db twice. The field can be removed and calls replaced by the protected method getDatabase()

File: server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1203_clean_orphelin_rows_in_resource_index.rb (hunk 2 above)
Comment (simonbrandhof) [inline server/sonar-web/src/main/webapp/WEB-INF/db/migrate/1203_clean_orphelin_rows_in_resource_index.rb]: oh got it, I rely on the fact that UUIDs are null.

**Comment by simonbrandhof** [comment]
After db migration (from 5.5 in my case), searching for a file in the search engine fails with
```
2016.06.09 12:55:45 INFO  app[o.s.p.m.Monitor] Process[ce] is up
2016.06.09 12:56:10 ERROR web[o.s.s.ui.JRubyFacade] Fail to render: http://localhost:9000/dashboard/index?id=com.sonarsource.cobol%3Asonar-cobol-plugin%3Asrc%2Fmain%2Fjava%2Fcom%2Fsonarsource%2Fcobol%2Fplugin%2FCobolPlugin.java
undefined local variable or method `page_index' for #<SonarAuthorizer:0x316fdc68>
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/lib/authorization/default_authorizer.rb:55:in `has_role_for_resources?'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/lib/need_authorization.rb:92:in `has_role_for_resources?'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/lib/need_authorization.rb:80:in `has_role_for_resource?'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/lib/need_authorization.rb:75:in `has_role?'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/lib/need_authorization.rb:165:in `has_role?'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/app/controllers/dashboard_controller.rb:218:in `load_resource'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/app/controllers/dashboard_controller.rb:28:in `index'
	org/jruby/RubyKernel.java:2223:in `send'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/base.rb:1333:in `perform_action'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/filters.rb:617:in `call_filters'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/filters.rb:610:in `perform_action_with_filters'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/benchmarking.rb:68:in `perform_action_with_benchmark'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/activesupport-2.3.15/lib/active_support/core_ext/benchmark.rb:17:in `ms'
	jar:file:/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/lib/server/jruby-complete-1.7.9.jar!/META-INF/jruby.home/lib/ruby/1.8/benchmark.rb:308:in `realtime'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/activesupport-2.3.15/lib/active_support/core_ext/benchmark.rb:17:in `ms'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/benchmarking.rb:68:in `perform_action_with_benchmark'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/rescue.rb:160:in `perform_action_with_rescue'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/flash.rb:151:in `perform_action_with_flash'
	org/jruby/RubyKernel.java:2223:in `send'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/base.rb:532:in `process'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/filters.rb:606:in `process_with_filters'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/base.rb:391:in `process'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/base.rb:386:in `call'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/routing/route_set.rb:450:in `call'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/dispatcher.rb:87:in `dispatch'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/dispatcher.rb:121:in `_call'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/actionpack-2.3.15/lib/action_controller/dispatcher.rb:130:in `build_middleware_stack'
	org/jruby/RubyProc.java:290:in `call'
	org/jruby/RubyProc.java:224:in `call'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/activerecord-2.3.15/lib/active_record/query_cache.rb:29:in `call'
	/Users/sbrandhof/dev/core/sonarqube/sonar-application/target/sonarqube-6.0-SNAPSHOT/web/WEB-INF/gems/gems/activerecord-2.3.15/lib/active_record/connection_adapters/abstract/query_cache.rb:34:in `cache'
```

**Comment by simonbrandhof** [comment]
LGTM

**Comment by sns-seb** [comment]
@simonbrandhof feedback applied, wanting for @eric-duquesnoy green light to merge into master

**Pull request merged**